day22/permutations.py

```diff
@@ -0,0 +1,20 @@
+def permutations(arr):
+    allPermutations = list()
+
+    if len(arr) <= 1:
+        return [arr]
+    else:
+        for i in xrange(len(arr)):
+            for permutation in permutations(arr[1:]):
+                # when i is 0, arr[0:1] is at the beginning of this list
+                allPermutations.append(permutation[:i] + arr[0:1] + permutation[i:])
+
+    return sorted(allPermutations)
```

Comment:
can't do a set because lists aren't hashable. Sorting output in the test sounds good